### DISCUSSION TITLE
Using generated api/models with kickstarter tools

### DISCUSSION THREAD

**Discussion by thisarattr** [kobylynskyi/graphql-java-codegen#797] [Q&A] [answered]
Following in [this](https://github.com/kobylynskyi/graphql-java-codegen/discussions/792#discussioncomment-1276146) answer, I was able to generate resolvers with `CompletableFuture<>` return type

but I have question related to this. Sorry im a newbie so most prob im missing something.
Consider eg below,
```
type Query {
   allUsers(first: Int, last: Int): UsersConnection
   user(id: ID): User
}
type User {
    id: ID!
    email: String
}
type UsersConnection {
    edges: [UsersEdge]
    totalCount: Int
    users: [User]
}
type UsersEdge {
    node: User
    cursor: String!
}
```
Generated model for UserConnection;
```
@javax.annotation.processing.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-09-05T15:20:22+1000"
)
public class UsersConnection implements java.io.Serializable {
    private java.util.List<UsersEdge> edges;
    private Integer totalCount;
    private java.util.List<User> users;
```
Generated resolver for User;
```
@Override
public CompletableFuture<User> user(String id, DataFetchingEnvironment env) throws Exception {
}
```

When I generate api with `CompletableFuture` return type the  generation works fine, but the kickstart schema scanner get confused that  UserResolver return  type `CompletableFuture<User>` but the  `allUsers` query returns a `UserConnection` object with just a `User` object

```
Caused by: graphql.kickstart.tools.SchemaClassScannerError: Two different classes used for type User:
- class java.util.concurrent.CompletableFuture:
|   return type of method public java.util.concurrent.CompletableFuture com.graphql.resolvers.$UserResolverDefinition$Intercepted.user(java.lang.String,graphql.schema.DataFetchingEnvironment)

- class com.graphql.model.User:
|   return type of method public java.util.List com.graphql.model.UserConnection.getUsers()
```

Have you come across this problem? is there any solution/work around for this from code generation perspective?

**Comment by kobylynskyi** [comment]
Is this related to [relay](https://www.graphql-java-kickstart.com/tools/relay/)?
If yes, the this could be achieved via [relayConfig](https://github.com/kobylynskyi/graphql-java-codegen/blob/master/docs/codegen-options.md#option-relayconfig)

**Comment by thisarattr** [comment, marked as answer]
Thanks for take a look at this 🙏 
I think I figured it out, dont think its related to relay 
Kickstarter schema parser option has way to resolve this conflict using the below flag. My issue got resolved with this flag, it uses resolver type and ignore the model
```
preferGraphQLResolver: In cases where you have a Resolver class and legacy class that conflict on type arguments, use the Resolver class instead of throwing an error. Specifically this situation can occur when you have a graphql schema type Foo with a bars property and classes:

// legacy class you can't change
class Foo {
  Set<Bar> getBars() {
    //...returns set of bars...
  }
}

// nice resolver that does what you want
class FooResolver implements GraphQLResolver<Foo> {
  Set<BarDTO> getBars() {
    // ...converts Bar objects to BarDTO objects and returns set...
  }
}
```
